tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/builder/IncrementalBuilderCancellationTest.xtend

```diff
@@ -219,4 +236,99 @@ class IncrementalBuilderCancellationTest extends AbstractIncrementalBuilderTest
 		assertEquals("(Error, [1:24 - 1:26], Couldn't resolve reference to IdentifiableElement 'a3'.)",
 			issues.sort.join(Strings.newLine))
 	}
+
+	/**
+	 * This test case tests a very specific edge case: A resource is 
+	 * (1) created,
+	 * (2) a cancellation happens, and
+	 * (3) the resource is removed again.
```

Comment:
maybe: "... is removed again _before the next incremental build iteration_." ? (or something similar)